config/src/main/java/org/springframework/security/config/annotation/web/configurers/RememberMeConfigurer.java

```diff
@@ -434,6 +434,9 @@ private UserDetailsService getUserDetailsService(H http) {
 	 * @return the remember me key to use
 	 */
 	private String getKey() {
+		if (this.key == null && this.rememberMeServices instanceof AbstractRememberMeServices) {
```

Comment:
Please move this check into the `null` check bellow, since we are already checking if `this.key == null`.